plugins/chilli/chilli_sessions_

```diff
@@ -0,0 +1,117 @@
+#!/bin/sh
+# -*- sh -*-
+
+: << =cut
+
+=head1 NAME
+
+chilli_sessions_ - Wildcard-plugin to monitor sessions state on Coova Chilli.
+
+=head1 CONFIGURATION
+
+This plugin does not normally require configuration.
+
+The plugin may need to run as root. This is configured like this:
+
+  [chilli_sessions_*]
+      user root
+
+This is a wildcard plugin. To monitor an instance, link
+chilli_sessions_<instance> to this file. For example :
+
+  ln -s /usr/share/munin/plugins/chilli_sessions_ \
+        /etc/munin/plugins/chilli_sessions_hotspot1
+
+will monitor hotspot1.
+
+For monitor all instances use :
+
+  ln -s /usr/share/munin/plugins/chilli_sessions_ \
+        /etc/munin/plugins/chilli_sessions_total
+
+=head1 AUTHOR
+
+OPENevents - Guillaume Marsay <guillaume.marsay@openevents.fr>
+
+=head1 LICENSE
+
+GPLv2
+
+=head1 MAGIC MARKERS
+
+ #%# family=auto
+ #%# capabilities=autoconf suggest
+
+=cut
+
+
+INSTANCE="${0##*chilli_}"
+CHILLI_PATH_BIN="/usr/sbin/chilli_query"
+CHILLI_PATH_SOCK="/var/run"
+
+
+case "$1" in
+  autoconf)
+    if [ -r "$CHILLI_PATH_BIN" ]; then
+      if [ "$INSTANCE" = "total" ]; then
+        echo "yes"
+        exit 0
+      else
+        if [ -r $CHILLI_PATH_SOCK/chilli_"$INSTANCE".sock ]; then
+          echo "yes"
+          exit 0
+        else
+          echo "no ($CHILLI_PATH_SOCK/chilli_$INSTANCE.sock not found)"
+          exit 0
+        fi
+      fi
+    else
+      echo "no ($CHILLI_PATH_BIN not found)"
+      exit 0
+    fi
+    ;;
+  suggest)
+      INSTANCES_LIST=$(ls /var/run/chilli_*.sock)
```

Comment:
This statement will not work with sockets containing a space. Additionally it neglects "CHILLI_PATH_SOCK".

How about this one?
```
find "$CHILLI_PATH_SOCK/" -name "chilli_*.sock" | while read filename; do
    basename "$file" .sock | cut -d _ -f 2
done
```
This should also take care for `shellcheck` warnings regarding quoting.